### ISSUE TITLE
An alternative way of testing videos

### ISSUE THREAD

**Issue by leotrs** [ManimCommunity/manim#353]
I was just talking to @yoshiask about testing the SVG branch and this came up. I think it's general enough so I'll share with the broader community for comments.

Proposal to test rendering:
1. Add a shape to the scene, say a circle with radius 1 and centered at the origin. Set the circle's color to pure red (#FF0000)
2. Render the scene
3. Get the camer's `pixel_array`. From that array, get a pixel that is inside the circle.
4. If the pixel's value is exactly equal to (255, 0, 0), the test passes.

Currently, we are trying to render the scene and output it to disk, and then comparing the output files. This is testing the rendering AS WELL AS the operation of _writing the file_. In general, we should have separate tests for these two operations.

Note the proposed method can be done with `--dry_run`, no output files necessary. So we can forget about all of the tests_cache, tests_data, etc. 

cc @huguesdevimeux as Tests Czar.

**Comment by huguesdevimeux** [comment]
> * Get the camer's `pixel_array`. From that array, get a pixel that is inside the circle.
> * If the pixel's value is exactly equal to (255, 0, 0), the test passes.

Maybe I didn't understood well but this is EXACTLY what is done for the graphical units tests.

**Comment by leotrs** [comment]
Hmmmm ok maybe this is less novel than I thought... but I think there is a major difference: the current tests render a whole scene and they need a cached frame to compare it to. What I'm suggesting is that we _also_ do this on the fly, checking for _some_ properties of _some_ pixels. This doesn't require cached frames.

I guess the real suggestion here is that we should write more tests :sweat_smile: (and not that we came up with a whole new way of doing things). Sorry, I got carried away!
